src/base/utils/os.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2018  Vladimir Golovnev <glassez@yandex.ru>
+ * Copyright (C) 2014  sledgehammer999 <sledgehammer999@qbittorrent.org>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#pragma once
+
+#include <QtGlobal>
+
+#ifdef Q_OS_WIN
+#include <windows.h>
+#if (NTDDI_VERSION >= NTDDI_VISTA)
+#define WIN_USE_DEFAULT_PROGRAMS
+#endif
+#endif
+
+namespace Utils
+{
+    namespace OS
+    {
+#if defined(Q_OS_WIN)
+        QString getPythonPath();
+
+        bool isAssociationsSet() noexcept;
+        void setAssociations();
+        bool isTorrentFileAssocSet();
+        bool isMagnetLinkAssocSet();
+        void setTorrentFileAssoc(bool value);
```

Comment:
Please explain what is intention behind `setXXXAssoc(false)`. From my point of view such call is a way to a confusion. Example: Prog1 is associated with file a type. Then Prog2 is installed and takes over the association. Now I want to set the associations back to Prog1, and do that via a system control module or Prog1 setting. Then I need nothing in the Prog2 to reset the association. Additionally, is it correct that clearing the settings checkbox in qBt will clear the association even if it is set to another program?